src/openforms/contrib/haal_centraal/tests/test_integration.py

```diff
@@ -0,0 +1,147 @@
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+from django.test import TestCase
+
+from openforms.authentication.constants import AuthAttribute
+from openforms.authentication.utils import store_auth_details, store_registrator_details
+from openforms.config.models import GlobalConfiguration
+from openforms.forms.tests.factories import FormStepFactory
+from openforms.prefill import prefill_variables
+from openforms.submissions.tests.factories import SubmissionStepFactory
+from openforms.typing import JSONValue
+from openforms.utils.tests.vcr import OFVCRMixin
+
+from ..models import HaalCentraalRequestOptions
+from .factories import HaalCentraalConfigFactory
+
+CONFIGURATION = {
+    "display": "form",
+    "components": [
+        {
+            "key": "bsn",
+            "type": "textfield",
+            "label": "BSN",
+            "prefill": {
+                "plugin": "haalcentraal",
+                "attribute": "burgerservicenummer",
+            },
+            "multiple": False,
+        },
+        {
+            "key": "address",
+            "type": "textfield",
+            "label": "Address",
+            "prefill": {
+                "plugin": "haalcentraal",
+                "attribute": "adressering",
+            },
+            "multiple": False,
+        },
+    ],
+}
+
+
+class BRPIntegrationTest(OFVCRMixin, TestCase):
+    """An integration test against a test instance, to make sure headers are set correctly.
+    The prefill feature is being used here, although any other feature making use of the Haal Centraal API
+    could have been used.
+    """
```

Comment:
Please format your docstrings with the [PEP-257](https://peps.python.org/pep-0257/) convention

Most notably, a single long line that is wrapping is too much, break it up into a short summary line, a blank line and then more elaborate descriptions:

```py
"""
Run full integration tests against a real API instance.

These tests ensure the correct behaviour of our implementation using a real API as reference. HTTP 
traffic is recorded using VCR... etc.
"""
```

The remark at the class level about headers being tested is also in the wrong location - you can add other tests/assertions to this case that test things *other* than the headers, so that part of the docstring belongs with the test itself.